### DISCUSSION TITLE
Problems debugging M1 (Silicon) and entering a character

### DISCUSSION THREAD

**Discussion by jesusdeveloper** [microsoft/vscode-cpptools#8072] [Q&A]
Hi!

I am trying to debug a program in C ++. As computers with an M1 processor are incompatible with gdb so I installed CodeLLDB.
So I debug the program and it seems to work, but when I have to read an input character (I set external terminal) Visual studio code directs me to a file with the memory area even though I enter the value in the terminal.

Could someone help me and tell me why this is happening please?

Thank you!

<img width="1378" alt="Captura de pantalla 2021-08-30 a las 1 35 59" src="https://user-images.githubusercontent.com/17848602/131268964-3e12b20f-632e-406e-97bf-1a50b42eebbc.png">

**Comment by bobbrow** [comment]
If you are having issues with CodeLLDB, you might also want to try asking in their repository: https://github.com/vadimcn/vscode-lldb/issues

**Comment by kinkard** [comment]
@jesusdeveloper Terminal input works fine with default settings:
<img width="769" alt="Screenshot 2021-08-31 at 00 08 13" src="https://user-images.githubusercontent.com/10885920/131405886-d88ea398-d8a0-48c9-9fb0-736c5272bac0.png">

My launch.json:
```json
{
  "version": "0.2.0",
  "configurations": [
    {
      "type": "lldb",
      "request": "launch",
      "name": "Debug",
      "program": "${workspaceFolder}/build/cpp_test",
      "args": [],
      "cwd": "${workspaceFolder}"
    }
  ]
}
```

Maybe you pressed f11 or some 'view in disassembly' button?

**Reply by jesusdeveloper** [reply]
Thanks my friends! I've gotten that it's works!!
The problem was a line "lldb.dereferencePointers": true" and this line hidden the breakpoints, I delete it and now it works!
Thank you my friends!!
